### DISCUSSION TITLE
in `login!': undefined method `split' for nil:NilClass (NoMethodError)

### DISCUSSION THREAD

**Discussion by AmmarKhawaja** [timdorr/tesla-api#523] [Q&A]
Hi, Im a bit new to Ruby and trying to use this API. Im getting this error:

in `login!': undefined method `split' for nil:NilClass (NoMethodError)
```ruby
      cookie = response.headers['set-cookie'].split(' ').first
                                             ^^^^^^
```
My code is just the example code in the README:

```ruby
require 'tesla_api'

tesla_api = TeslaApi::Client.new(email: "email", client_id: "id", client_secret: "secret")
tesla_api.login!('password123')
# Or if you have an access token:
#tesla_api = TeslaApi::Client.new(access_token: access_token) 

model_s = tesla_api.vehicles.first # => <TeslaApi::Vehicle>

model_s.wake_up
model_s.auto_conditioning_start unless model_s.climate_state["is_auto_conditioning_on"]

model_s.set_charge_limit(90)
model_s.charge_start

charge_state = model_s.charge_state
puts "Your Model S is #{charge_state["charging_state"]} " +
     "with a SOC of #{charge_state["battery_level"]}% " + 
     "and an estimate range of #{charge_state["est_battery_range"]} miles"
```

Any guidance would be helpful. Thank you

**Comment by dewski** [comment]
I am running into this as well with just the first 3 lines of the provided code above.

**Comment by duizendnegen** [comment]
Unfortunately, as described in https://tesla-api.timdorr.com/api-basics/authentication, the method of authentication you're using is deprecated. You can try to emulate the browser using Puppeteer for authentication, as detailed in https://github.com/timdorr/tesla-api/discussions/501#discussioncomment-1829937.

**Comment by lemeo** [comment]
Hey @timdorr, I got the same issue. You already updated this ruby lib with the new SSO authentication way.
https://github.com/timdorr/tesla-api/blob/5e23c3bf856b62a6eb931d4cae717516186bb7b7/lib/tesla_api/client.rb#L61-L146
 
Based on your update, from my understanding, the step to authenticate is as follows:
Step 1:
```ruby
tesla_api = TeslaApi::Client.new(email: "tesla account email", client_id: nil, client_secret: nil)
```
We do not need to set client_id and client_secret since it is deprecated, and we have the default `client_id` set as `ownerapi` within `login!(password)` function.  
Step 2:
```ruby
tesla_api.login!('tesla account email')
```

Why do I get this error?
```
NoMethodError: undefined method `split' for nil:NilClass
```
Do we need to update the ruby lib? Or let me know the latest way to use this Ruby lib to authenticate. Thank you!
